src/domain_bridge/domain_bridge.cpp

```diff
@@ -185,6 +186,46 @@ class DomainBridgeImpl
       {
         const std::string & type = topic_bridge.type_name;
 
+        // Apply QoS overrides
+        const auto & qos_options = topic_options.qos_options();
+        rclcpp::QoS qos(qos_options.depth());
+        qos.history(qos_options.history());
+        if (qos_options.reliability()) {
+          qos.reliability(qos_options.reliability().value());
+        } else {
+          qos.reliability(qos_match.qos.reliability());
+        }
+        if (qos_options.durability()) {
+          qos.durability(qos_options.durability().value());
+        } else {
+          qos.durability(qos_match.qos.durability());
+        }
+        if (qos_options.deadline()) {
+          const auto deadline_ns = qos_options.deadline().value();
+          if (deadline_ns < 0) {
+            qos.deadline(
+              rclcpp::Duration::from_nanoseconds(std::numeric_limits<std::int64_t>::max()));
+          } else {
+            qos.deadline(rclcpp::Duration::from_nanoseconds(deadline_ns));
+          }
+        } else {
+          qos.deadline(qos_match.qos.deadline());
+        }
+        if (qos_options.lifespan()) {
+          const auto lifespan_ns = qos_options.lifespan().value();
+          if (lifespan_ns < 0) {
+            qos.lifespan(
+              rclcpp::Duration::from_nanoseconds(std::numeric_limits<std::int64_t>::max()));
```

Comment:
maybe:

```suggestion
            qos.lifespan(RMW_DURATION_INFINITE);
```

we should have something like `rclcpp::Duration()::Infinity()` in rclcpp though.

I'm not sure if your code above does exactly the same, thought it should.

Reply:
actually, maybe the default value should be `RMW_DURATION_INFINITE` (instead of `-1`) and the `if (lifespan_ns < 0)` is never needed (?)

Reply:
I wanted an easy way for users to explicitly specify an infinite duration in YAML (hence negative values), though in retrospect we could also use a string `infinite` or `max`.